internal/proxy/search_pipeline.go

```diff
@@ -592,6 +598,154 @@ func (op *filterFieldOperator) run(ctx context.Context, span trace.Span, inputs
+		Tasks: lo.Map(op.tasks, func(task *highlightTask, _ int) *querypb.HighlightTask { return task.HighlightTask }),
+	}
+
+	for _, task := range req.GetTasks() {
+		textFieldDatas, ok := lo.Find(datas, func(data *schemapb.FieldData) bool { return data.FieldId == task.GetFieldId() })
+		if !ok {
+			return nil, errors.Errorf("get highlight failed, text field not in output field %s: %d", task.GetFieldName(), task.GetFieldId())
+		}
+		texts := textFieldDatas.GetScalars().GetStringData().GetData()
+		task.Texts = append(task.Texts, texts...)
+		task.CorpusTextNum = int64(len(texts))
+		field, ok := lo.Find(op.fieldSchemas, func(schema *schemapb.FieldSchema) bool {
+			return schema.GetFieldID() == task.GetFieldId()
+		})
+		if !ok {
+			return nil, errors.Errorf("get highlight failed, field not found in schema %s: %d", task.GetFieldName(), task.GetFieldId())
+		}
+
+		// if use multi analyzer
+		// get analyzer field data
+		helper := typeutil.CreateFieldSchemaHelper(field)
+		if v, ok := helper.GetMultiAnalyzerParams(); ok {
+			params := map[string]any{}
+			err := json.Unmarshal([]byte(v), &params)
+			if err != nil {
+				return nil, errors.Errorf("get highlight failed, get invalid multi analyzer params-: %v", err)
+			}
+			analyzerField, ok := params["by_field"]
+			if !ok {
+				return nil, errors.Errorf("get highlight failed, get invalid multi analyzer params, no by_field")
+			}
+
+			analyzerFieldDatas, ok := lo.Find(datas, func(data *schemapb.FieldData) bool { return data.FieldName == analyzerField.(string) })
+			if !ok {
+				return nil, errors.Errorf("get highlight failed, analyzer field not in output field")
+			}
+			task.AnalyzerNames = append(task.AnalyzerNames, analyzerFieldDatas.GetScalars().GetStringData().GetData()...)
+		}
+	}
+
+	task := &HighlightTask{
+		ctx:                 ctx,
+		lb:                  op.lbPolicy,
+		Condition:           NewTaskCondition(ctx),
+		GetHighlightRequest: req,
+		collectionName:      op.collectionName,
+		collectionID:        op.collectionID,
+		dbName:              op.dbName,
+	}
+	if err := op.scheduler.dqQueue.Enqueue(task); err != nil {
+		return nil, err
+	}
+
+	if err := task.WaitToFinish(); err != nil {
+		return nil, err
+	}
+
+	rowNum := len(result.Results.GetScores())
+	HighlightResults := []*commonpb.HighlightResult{}
+	if rowNum != 0 {
+		rowDatas := lo.Map(task.result.Results, func(result *querypb.HighlightResult, i int) *commonpb.HighlightData {
+			return buildStringFragments(op.tasks[i/rowNum], i%rowNum, result.GetFragments())
+		})
+
+		for i, task := range req.GetTasks() {
+			HighlightResults = append(HighlightResults, &commonpb.HighlightResult{
+				FieldName: task.GetFieldName(),
+				Datas:     rowDatas[i*rowNum : (i+1)*rowNum],
+			})
+		}
+	}
+
+	result.Results.HighlightResults = HighlightResults
+	return []any{result}, nil
+}
+
+func buildStringFragments(task *highlightTask, idx int, frags []*querypb.HighlightFragment) *commonpb.HighlightData {
```

Comment:
put this in a seperate fuction too.

we might need different policy to build string fragments